spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/MaskHead.scala

```diff
@@ -0,0 +1,175 @@
+import com.intel.analytics.bigdl.utils.{T, Table}
+
+class MaskHead(
+  val inChannels: Int,
+  val resolution: Int,
+  val scales: Array[Float],
+  val samplingRratio: Float,
+  val layers: Array[Int],
+  val dilation: Int,
+  val numClasses: Int,
+  val useGn: Boolean = false)(implicit ev: TensorNumeric[Float])
+  extends BaseModule[Float] {
+
+  override def buildModel(): Module[Float] = {
+    val featureExtractor = this.maskFeatureExtractor(
+      inChannels, resolution, scales, samplingRratio, layers, dilation, useGn)
+    val dimReduced = layers(layers.length - 1)
+    val predictor = this.maskPredictor(dimReduced, numClasses, dimReduced)
+    val postProcessor = new MaskPostProcessor()
+
+    /**
+     * input: feature-maps from possibly several levels and proposal boxes
+     * return:
+     * first tensor: the result of the feature extractor
+     * second tensor: proposals (list[BoxList]): during training, the original proposals
+     *      are returned. During testing, the predicted boxlists are returned
+     *      with the `mask` field set
+     */
+    val features = Input()
+    val proposals = Input()
+    val labels = Input()
+
+    val maskFeatures = featureExtractor.inputs(features, proposals)
+    val maskLogits = predictor.inputs(maskFeatures)
+    val result = postProcessor.inputs(maskLogits, proposals, labels)
+
+    Graph(Array(features, proposals, labels), Array(maskFeatures, result))
+  }
+
+  private[nn] def maskPredictor(inChannels: Int,
+                                numClasses: Int,
+                                dimReduced: Int): Module[Float] = {
+    val convMask = SpatialFullConvolution(inChannels, dimReduced,
+      kW = 2, kH = 2, dW = 2, dH = 2)
+    val maskLogits = SpatialConvolution(nInputPlane = dimReduced,
+      nOutputPlane = numClasses, kernelW = 1, kernelH = 1, strideH = 1, strideW = 1)
+
+    // init weight & bias, Caffe2 implementation uses MSRAFill,
+    convMask.setInitMethod(MsraFiller(false), Zeros)
+    maskLogits.setInitMethod(MsraFiller(false), Zeros)
+
+    val model = Sequential[Float]()
+    model.add(convMask).add(ReLU[Float]()).add(maskLogits)
+    model
+  }
+
+  private[nn] def maskFeatureExtractor(inChannels: Int,
+                                       resolution: Int,
+                                       scales: Array[Float],
+                                       samplingRatio: Float,
+                                       layers: Array[Int],
+                                       dilation: Int,
+                                       useGn: Boolean = false): Module[Float] = {
+
+    require(dilation == 1, s"Only support dilation = 1, but get ${dilation}")
+
+    val model = Sequential[Float]()
+    model.add(Pooler(resolution, scales, samplingRatio.toInt))
+
+    var nextFeatures = inChannels
+    var i = 0
+    while (i < layers.length) {
+      val features = layers(i)
+      // todo: not support dilation convolution
```

Comment:
remove not ?

Reply:
and is there option for user to switch from spatial conv to dilaiton conv ?

Reply:
yeah, should remove not.
Actually, it is dilated conv in pytorch and dilation = 1 for MaskRCNN.
As our dilated conv must be with bias, so here I use normal conv and make a comment.